### PR TITLE
MLIBZ-2681: Migrate from GCM to FCM

### PR DESCRIPTION
**Pull request by romanogolikhin**
#### Description
This goal is to make the necessary client/SDK changes to support Firebase Cloud Messaging. 

#### Changes
New extension (FCMPushExtension)
New push class(FCMPush)
New abstract service(KinveyFCMService)

#### Tests
No tests.

### CODE REVIEW

File: Kinvey.Android/Push/KinveyFCMService.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        protected override void OnHandleIntent(Intent intent)
+        {
+
+            lock (LOCK)
+            {
+                if (sWakeLock == null)
+                {
+                    var pm = PowerManager.FromContext(this.ApplicationContext);
+                    sWakeLock = pm.NewWakeLock(WakeLockFlags.Partial, KINVEY_FCM);
```
Comment (heyzooi) [inline Kinvey.Android/Push/KinveyFCMService.cs]: @romanogolikhin why do we need to keep the device on while you are handling the intent?

Comment (heyzooi) [reply Kinvey.Android/Push/KinveyFCMService.cs]: right... worth it to do some research about it. @vinaygahlawat will check how this is done in the android sdk. hold on for now.

**Comment by heyzooi** [comment]
@romanogolikhin let's rebase this branch after merge #188 so the conflict with `Kinvey.TestApp.Shared/Pages/ContractsPage.xaml` and `Kinvey.TestApp.Shared/Pages/ContractsPage.xaml.cs` will be solved

**Comment by romanogolikhin** [comment]
@heyzooi I have rebased the branch. Conflicts disappeared.

**Pull request merged**